XYZDrivers/src/main/java/com/xyzdrivers/services/SQLService.java

```diff
@@ -0,0 +1,400 @@
+/**
+ * @file    jdbcDriver.java
+ * @author  alexander collins
+ * @created 02/11/2017
+ * @updated 03/11/2017, alexander collins
+ * @notes   - The private functions could probably be made public.
+ *          - Not tested
+ * @issues  line 71. getPrimaryKeyName() doesn't work, any function using it won't work either.
+ *          needs testing.
+ */
+package com.xyzdrivers.services;
+
+import java.sql.*;
+import java.util.*;
+
+public class SQLService
+{
+//variables
+    private Connection DB;
+    private DatabaseMetaData DBMetaData;
+    
+    private PreparedStatement statement;
+    private ResultSetMetaData resultsMetaData;
+    private ResultSet results;
+//constructors
+    /**
+     * Constructor for jdbcDriver.
+     * <code>dbConnection</code> is tested by retrieving <code>DatabaseMetaData</code> from the <code>Connection</code>.
+     * 
+     * @param dbConnection A <code>Connection</code> to the DB being accessed.
+     * 
+     * @throws SQLException
+     */
+    public SQLService(Connection dbConnection)
+            throws SQLException
+    {
+        //pass DB connection
+        DB = dbConnection;
+        
+        //get database metadata
+        DBMetaData = DB.getMetaData();
+    }
+    
+//private methods
+    private ResultSet executeQueryStatement(String sql, Object... parameters)
+            throws SQLException
+    {
+        //prepare statement
+        statement = DB.prepareStatement(sql);
+        for (int i = 0; i < parameters.length; i++)
+            statement.setObject(i + 1, parameters[i]);
+        
+        //execute statement
+        return statement.executeQuery();
+    }
+    
+    private int executeUpdateStatement(String sql, Object... parameters)
+            throws SQLException
+    {
+        //prepare statement
+        statement = DB.prepareStatement(sql);
+        for (int i = 0; i < parameters.length; i++)
+            statement.setObject(i + 1, parameters[i]);
+        
+        //execute statement
+        return statement.executeUpdate();
+    }
+    
+    /**
+     * @ISSUE NOT WORKING
```

Comment:
Does this not work?

Reply:
Yeah I remember when I was developing the file that I was using this function and for some reason it stopped working and I didn't have time to debug why.
Left it there along with the functions using it in case I (or someone else) got round to fixing it.